addons/account/models/product.py

```diff
@@ -53,6 +54,12 @@ def get_product_accounts(self, fiscal_pos=None):
             fiscal_pos = self.env['account.fiscal.position']
         return fiscal_pos.map_accounts(accounts)
 
+    @api.constrains('uom_id')
+    def _check_uom_not_in_invoice(self):
+        for template in self:
+            invoices = self.env['account.move.line'].sudo().search([('product_id.product_tmpl_id.id', '=', template.id)], limit=1)
+            if invoices:
+                raise ValidationError(_('The product "%s" is used in invoices. You cannot change its Unit of Measure.', self.display_name))
```

Comment:
it is not `self.display_name`, it is `template.display_name`.
@yosa-odoo

Reply:
Indeed, forgot to made the change when I changed to `for product in self`.
I'll stop the fw-port and make the change in 14